lib/spack/spack/url.py

```diff
@@ -199,6 +201,10 @@ def parse_version_offset(path):
         # https://github.com/hpc/mpileaks/releases/download/v1.0/mpileaks-1.0.tar.gz
         (r'github.com/[^/]+/[^/]+/releases/download/v?([^/]+)/.*$', path),
 
+        # e.g., https://gitlab.kitware.com/vtk/vtk/repository/archive.tar.bz2?ref=v7.0.0
+        # e.g., https://gitlab.kitware.com/icet/icet/repository/archive.tar.bz2?ref=IceT-2.1.1
```

Comment:
p.s. the point is, in one year or so nobody might remember the text in this PR which gives rationale. What will count are unit tests to make sure the feature works as expected and in-code comments to clarify what's happening.